Project/VetAppointmentAPI/VetAppointmentAPI/VetAppointment.Infrastructure/Repositories/Impl/OfficeRepository.cs

```diff
@@ -8,5 +8,18 @@ namespace VetAppointment.Application.Repositories.Impl
     public class OfficeRepository : BaseRepository<Office>, IOfficeRepository
     {
         public OfficeRepository(IDatabaseContext context) : base(context) { }
+
+        public async Task DeleteAsync(Office office)
```

Comment:
Again, let's move all these methods in the IBaseRepository and BaseRepository classes, and making them generics.